src/mongocxx/test/search_index_view.cpp

```diff
@@ -1 +1,211 @@
+
+    SECTION("create one with model") {
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
+        auto siv = coll.search_indexes();
+        // {
+        //   name: 'test-search-index',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name = "test-search-index";
+        auto definition = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model = search_index_model(name, definition.view());
+
+        REQUIRE(siv.create_one(model) == "test-search-index");
+
+        REQUIRE(wait_for_search_index(siv, model, true, false));
+
+        std::cout << "create one with model SUCCESS" << std::endl;
+    }
+
+    SECTION("create many") {
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
+        auto siv = coll.search_indexes();
+        // {
+        //   name: 'test-search-index-1',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name1 = "test-search-index-1";
+        auto definition1 = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model1 = search_index_model(name1, definition1.view());
+
+        // {
+        //   name: 'test-search-index-2',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name2 = "test-search-index-2";
+        auto definition2 = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model2 = search_index_model(name2, definition2.view());
+
+        std::vector<search_index_model> models = {model1, model2};
+
+        std::vector<std::string> result = siv.create_many(models);
+        std::vector<std::string> expected = {"test-search-index-1", "test-search-index-2"};
+        REQUIRE(result == expected);
+
+        REQUIRE(wait_for_search_index(siv, model1, true, false));
+        REQUIRE(wait_for_search_index(siv, model2, true, false));
+
+        std::cout << "create many SUCCESS" << std::endl;
+    }
+
+    SECTION("drop one") {
+        bsoncxx::oid id;
+        db.create_collection(id.to_string());
+        collection coll = db[id.to_string()];
+        auto siv = coll.search_indexes();
+        // {
+        //   name: 'test-search-index',
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name = "test-search-index";
+        auto definition = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model = search_index_model(name, definition.view());
+
+        REQUIRE(siv.create_one(model) == "test-search-index");
+
+        REQUIRE(wait_for_search_index(siv, model, true, false));
+
+        siv.drop_one(name);
+
+        REQUIRE(wait_for_search_index(siv, model, false, false));
```

Comment:
The spec prose tests say to check for empty results:

> #. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until ``listSearchIndexes`` returns an empty array.

As implemented, `does_search_index_exist_on_cursor` only returns `true` if some indexes are returned. I expect the check may time out if the results are empty before they change state.

Consider replacing `wait_for_search_index` with a more generic `assert_soon` helper to separate the logic to wait for a condition:

```c++
// `assert_soon` repeatedly calls `fn` and asserts `fn` eventually returns true.
// `fn` is called every five seconds. Fails if `fn` does not return true within five minutes.
void assert_soon(std::function<bool()> fn) {
    auto start = std::chrono::high_resolution_clock::now();
    while (std::chrono::high_resolution_clock::now() - start < std::chrono::minutes(5)) {
        if (fn()) {
            return;
        }
        std::this_thread::sleep_for(std::chrono::seconds(5));
    }

    FAIL("Expected function to return true within five minutes, but did not");
}
```

Then, the "wait for empty" condition could be implemented as:

```c++
assert_soon([&siv](void) -> bool {
    auto cursor = siv.list();
    // Return true if empty results are returned.
    return cursor.begin() == cursor.end();
});
```